src/librustc/session/mod.rs

```diff
@@ -336,6 +336,10 @@ impl NodeIdAssigner for Session {
     fn peek_node_id(&self) -> NodeId {
         self.next_node_id.get().checked_add(1).unwrap()
     }
+
+    fn diagnostic(&self) -> &errors::Handler {
+        self.diagnostic()
```

Comment:
yeah, inherent has precedence.

and looking more at this has made me realized that one cannot resort to UFCS as a way to try to "clarify" that the inherent method is what is intended. There's no way to say "I want an inherent impl and only an inherent impl", at least not via UFCS...